### PR TITLE
add outline for configuration

### PR DESCRIPTION
**Pull request by kevodwyer**

### CODE REVIEW

File: src/main/java/org/peergos/HostBuilder.java [hunk 1]
```diff
@@ -82,12 +83,33 @@ public HostBuilder generateIdentity() {
...
+
+
+
     public HostBuilder setPrivKey(PrivKey privKey) {
         this.privKey = privKey;
         this.peerId = PeerId.fromPubKey(privKey.publicKey());
         return this;
     }
 
+    public static HostBuilder build(String privKeyBase64, int listenPort, ProviderStore providers, RecordStore records, Blockstore blocks) {
```
Comment (ianopolous) [inline src/main/java/org/peergos/HostBuilder.java:99]: Why do we need this method with hard coded protocols?

File: src/main/java/org/peergos/util/InstanceAdmin.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.peergos.cbor.Cborable;
+
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.CompletableFuture;
+
+
+public interface InstanceAdmin {
+
+    CompletableFuture<VersionInfo> getVersionInfo();
```
Comment (ianopolous) [inline src/main/java/org/peergos/util/InstanceAdmin.java:13]: Why do we need this class? What does kubo do for version, is it json?

File: src/main/java/org/peergos/HostBuilder.java (hunk 1 above)
Comment (kevodwyer) [reply src/main/java/org/peergos/HostBuilder.java:107]: copied relevant code to Server.java

File: src/main/java/org/peergos/util/InstanceAdmin.java (hunk 2 above)
Comment (kevodwyer) [reply src/main/java/org/peergos/util/InstanceAdmin.java]: removed class

File: src/main/java/org/peergos/HostBuilder.java (hunk 1 above)
Comment (ianopolous) [reply src/main/java/org/peergos/HostBuilder.java:107]: So this method can go?

File: src/main/java/org/peergos/net/APIHandler.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                case ID: { // https://docs.ipfs.tech/reference/kubo/rpc/#api-v0-id
+                    PeerId peerId = node.getPeerId();
+                    Map res = new HashMap<>();
+                    res.put("ID",  peerId.toBase58());
+                    replyJson(httpExchange, JSONParser.toString(res));
+                    break;
+                }
+                case VERSION: { // https://docs.ipfs.tech/reference/kubo/rpc/#api-v0-version
+                    Map res = new HashMap<>();
+                    res.put("Version", "0.4.11");
```
Comment (ianopolous) [inline src/main/java/org/peergos/net/APIHandler.java:75]: Why this version?

File: src/main/java/org/peergos/util/Config.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package org.peergos.util;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Config {
+    private final Map configuration;
```
Comment (ianopolous) [inline src/main/java/org/peergos/util/Config.java:9]: Can we enforce that this is a Map<String, Object> or better?

File: src/main/java/org/peergos/net/APIHandler.java (hunk 3 above)
Comment (kevodwyer) [reply src/main/java/org/peergos/net/APIHandler.java:75]: That is the version that the java ipfs client checks for as part of establishing connection. Happy to change it

File: src/main/java/org/peergos/util/Config.java (hunk 4 above)
Comment (kevodwyer) [reply src/main/java/org/peergos/util/Config.java]: done

File: src/main/java/org/peergos/net/APIHandler.java (hunk 3 above)
Comment (ianopolous) [reply src/main/java/org/peergos/net/APIHandler.java:75]: I would put in our version here

File: src/main/java/org/peergos/HostBuilder.java (hunk 1 above)
Comment (kevodwyer) [reply src/main/java/org/peergos/HostBuilder.java]: yes, done

File: src/main/java/org/peergos/net/APIHandler.java (hunk 3 above)
Comment (kevodwyer) [reply src/main/java/org/peergos/net/APIHandler.java]: done

**Review by ianopolous — APPROVED** [review]

**Pull request merged**